astropy/units/__init__.py

```diff
@@ -10,44 +10,56 @@
 code under a BSD license.
 """
 
-# Import order matters here -- circular dependencies abound!
-# Lots of things to import - go from more basic to advanced, so that
-# whatever advanced ones need generally has been imported already;
-# this also makes it easier to understand where most time is spent
```

Comment:
I like your reordering, but we may be losing this advantage of understanding where most time is spent - this was added in ff3ee7ec179cdbbc9574e5f52b45f49aba82deac and c39967d865a332ea875416705ffcd307db532929, and expanded on in cf71e00be762ce867d68d4ca69b9054647f0f99e

Reply:
I can remove. I thought it was useful to get closer to a full automatic import sorting.
Also, I think `units` is a prime candidate for  #13821, which will also change the nature of import time.
Thoughts?

Reply:
Hmm, tricky.  On the one hand good to keep the PR focussed on getting `__all__` set up, and do the sorting later when it is actually useful, on the other I actually quite liked in your commit that now we know that for a whole slew of imports, the order does not actually matter. On balance, let's go with what you have!